includes/abstracts/abstract-wc-stripe-payment-gateway.php

```diff
@@ -936,7 +936,13 @@ public function add_payment_method() {
 
 		$source = ! empty( $_POST['stripe_source'] ) ? wc_clean( $_POST['stripe_source'] ) : '';
 
-		$source_object = WC_Stripe_API::retrieve( 'sources/' . $source );
+		// Special handling for Stripe's `payment_method`s when adding a new payment method.
+		$source_object = null;
+		if ( is_add_payment_method_page() && 0 === strpos( $source, 'pm' ) ) {
+			$source_object = WC_Stripe_API::retrieve( 'payment_methods/' . $source );
+		} else {
+			$source_object = WC_Stripe_API::retrieve( 'sources/' . $source );
+		}
```

Comment:
I was experimenting with using the PaymentMethod returned from the SetupIntent as a direct replacement for the Source. Didn't work, the site just crashed 🤷‍♂️ .

If there is a way to coerce the PaymentMethod into a Source, then this would be the place to do so.

Reply:
That's a completely different topic and is not something we can do without fully switching to the Payment Methods API.

Reply:
Agreed, I just wanted to mention what this code was since I pushed the PR in a very messy state. Definitely safe to remove this 😅